lib/galaxy/model/base.py

```diff
@@ -41,6 +42,25 @@
 REQUEST_ID = ContextVar("request_id", default=_request_state.copy())
 
 
+def commit(session: Union[scoped_session, Session, "SessionlessContext"]):
+    """
+    Wraps SQLAlchemy's Session.commit method. Should be called from galaxy instead of using
+    SQLAlchemy's method directly.
+    """
+    err = None
+    try:
+        session.commit()
+    except PendingRollbackError as e:
+        err = e
+    except Exception as e:
+        err = e
+        raise
+    finally:
+        if err:
+            session.rollback()
+            log.error("Database transaction rolled back due to the following error: %s" % err)
```

Comment:
```suggestion
            log.error("Database transaction rolled back due to the following error: %s", err)
```

Reply:
If we want to do this we surely want to use log.exception ?

Reply:
Yes, I wasn't sure which one is preferable here. I'll make the change now regardless of whether we decide to go with this or not - just to make sure this suggestion is not lost.